components/BottomAppBar/src/private/MDCBottomAppBarView.m

```diff
@@ -0,0 +1,250 @@
+
+@end
+
+@implementation MDCBottomAppBarView
+
+- (void)setupBottomBarView {
+  self.cutView = [[MDCBottomAppBarCutView alloc] initWithFrame:self.bounds];
+  [self addSubview:_cutView];
+  self.autoresizingMask = (UIViewAutoresizingFlexibleWidth |
+                           UIViewAutoresizingFlexibleLeftMargin |
+                           UIViewAutoresizingFlexibleRightMargin);
+  self.layoutDirection = self.mdc_effectiveUserInterfaceLayoutDirection;
+  [self setFloatingButtonVisible:YES];
+  [self addBottomBarLayer];
+}
+
+- (void)addBottomBarLayer {
+  if (self.bottomBarLayer) {
+    [self.bottomBarLayer removeFromSuperlayer];
+  }
+  self.bottomBarLayer = [MDCBottomAppBarLayer layer];
+  [self.cutView.layer addSublayer:self.bottomBarLayer];
+}
+
+- (void)setFloatingButton:(MDCFloatingButton *)floatingButton {
+  _floatingButton = floatingButton;
+  [self.floatingButton sizeToFit];
+  [self insertSubview:self.floatingButton atIndex:0];
+}
+
+- (void)layoutSubviews {
+  [super layoutSubviews];
+  self.floatingButton.translatesAutoresizingMaskIntoConstraints = NO;
+  self.floatingButton.center =
+      [self getFloatingButtonCenterPositionForWidth:CGRectGetWidth(self.bounds)];
+  [self renderPathBasedOnFloatingButtonVisibility:NO];
+}
+
+- (void)renderPathBasedOnFloatingButtonVisibility:(BOOL)animated {
+  if (self.floatingButtonVisible) {
+    [self cutBottomAppBarView:animated];
+  } else {
+    [self healBottomAppBarView:animated];
+  }
+}
+
+- (CGPoint)getFloatingButtonCenterPositionForWidth:(CGFloat)width {
+
+  CGPoint floatingButtonPoint = CGPointZero;
+  CGFloat halfDefaultDimension = CGRectGetWidth(self.floatingButton.bounds) / 2;
+  CGFloat midX = width / 2;
+
+  switch (self.floatingButtonPosition) {
+    case MDCBottomAppBarFloatingButtonPositionLeading: {
+      if (self.layoutDirection == UIUserInterfaceLayoutDirectionLeftToRight) {
+        floatingButtonPoint = CGPointMake(kMDCBottomAppBarFloatingButtonPositionX,
+                                          halfDefaultDimension);
+      } else {
+        floatingButtonPoint = CGPointMake(width - kMDCBottomAppBarFloatingButtonPositionX,
+                                          halfDefaultDimension);
+      }
+      break;
+    }
+    case MDCBottomAppBarFloatingButtonPositionCenter: {
+      floatingButtonPoint = CGPointMake(midX, halfDefaultDimension);
+      break;
+    }
+    case MDCBottomAppBarFloatingButtonPositionTrailing: {
+      if (self.layoutDirection == UIUserInterfaceLayoutDirectionLeftToRight) {
+        floatingButtonPoint = CGPointMake(width - kMDCBottomAppBarFloatingButtonPositionX,
+                                          halfDefaultDimension);
+      } else {
+        floatingButtonPoint = CGPointMake(kMDCBottomAppBarFloatingButtonPositionX,
+                                          halfDefaultDimension);
+      }
+      break;
+    }
+    default:
+      break;
+  }
+  return floatingButtonPoint;
+}
+
+- (void)cutBottomAppBarView:(BOOL)animated {
```

Comment:
Per UIKIt, add Animated: to method name.

Reply:
Done.